neuland/executables/neulandAna.cxx

```diff
@@ -0,0 +1,129 @@
+    { { "neuland", "tacquila" },
+      []() { return Digitizing::CreateEngine(UsePaddle<NeulandPaddle>(), UseChannel<TacquilaChannel>()); } },
+    { { "mock", "tamex" },
+      []() { return Digitizing::CreateEngine(UsePaddle<MockPaddle>(), UseChannel<TamexChannel>()); } },
+    { { "neuland", "mock" },
+      []() { return Digitizing::CreateEngine(UsePaddle<NeulandPaddle>(), UseChannel<MockChannel>()); } },
+    { { "mock", "mock" },
+      []() { return Digitizing::CreateEngine(UsePaddle<MockPaddle>(), UseChannel<MockChannel>()); } }
+};
+
+namespace po = boost::program_options;
+
+int main(int argc, const char** argv)
+{
+    TStopwatch timer;
+    timer.Start();
+
+    auto channelName = std::string{ "tacquila" };
+    auto paddleName = std::string{ "neuland" };
+    auto paraFileName = std::string{ "para.root" };
+    auto simuFileName = std::string{ "simu.root" };
+    auto digiFileName = std::string{ "digi.root" };
+    auto logLevel = std::string{ "error" };
+    auto eventNum = 0;
+
+    auto desc = po::options_description{ "options for neuland data analysis" };
+    desc.add_options()("help", "help message")(
+        "paddle", po::value<std::string>(), R"(set the paddle name. e.g. "neuland")")(
+        "channel", po::value<std::string>(), R"(set the channel name. e.g. "tamex", "tacquila")")(
+        "simuFile", po::value<std::string>(), "set the base filename of simulation input")(
+        "digiFile", po::value<std::string>(), "set the base filename of digitization ouput")(
+        "paraFile", po::value<std::string>(), "set the base filename of parameter input")(
+        "eventN,n", po::value<int>(), "set the number of events")(
+        "logLevel,v", po::value<std::string>(), "set log level of fairlog");
+
+    auto varMap = po::variables_map{};
+    po::store(po::parse_command_line(argc, argv, desc), varMap);
+    po::notify(varMap);
+
+    if (varMap.count("help") != 0U)
+    {
+        std::cout << desc << std::endl;
+        return 0;
+    }
+    if (varMap.count("paddle") != 0U)
+    {
+        paddleName = varMap["paddle"].as<std::string>();
+    }
+    if (varMap.count("channel") != 0U)
+    {
+        channelName = varMap["channel"].as<std::string>();
+    }
+    if (varMap.count("simuFile") != 0U)
+    {
+        simuFileName = varMap["simuFile"].as<std::string>() + std::string{ ".root" };
+    }
+    if (varMap.count("digiFile") != 0U)
+    {
+        digiFileName = varMap["digiFile"].as<std::string>() + std::string{ ".root" };
+    }
+    if (varMap.count("paraFile") != 0U)
+    {
+        paraFileName = varMap["paraFile"].as<std::string>() + std::string{ ".root" };
+    }
+    if (varMap.count("logLevel") != 0U)
+    {
+        logLevel = varMap["logLevel"].as<std::string>();
+    }
+    if (varMap.count("eventN") != 0U)
+    {
+        eventNum = varMap["eventN"].as<int>();
+    }
+    FairLogger::GetLogger()->SetLogScreenLevel(logLevel.c_str());
+
+    auto run = new FairRunAna{};
```

Comment:
pointer?

Reply:
Yes, I just copied it from root Macro. Clang-tidy also report warnings for this.